vyper/compiler/output.py

```diff
@@ -108,9 +108,15 @@ def build_integrity(compiler_data: CompilerData) -> str:
 def build_external_interface_output(compiler_data: CompilerData) -> str:
     interface = compiler_data.annotated_vyper_module._metadata["type"].interface
     stem = PurePath(compiler_data.contract_path).stem
+
     # capitalize words separated by '_'
     # ex: test_interface.vy -> TestInterface
-    name = "".join([x.capitalize() for x in stem.split("_")])
+    def capitalize_part(part):
+        if len(part) > 0:  # safe even if 1 cause [1:] will be just empty string
+            return part[0].upper() + part[1:]
+        return ""
+
+    name = "".join([capitalize_part(x) for x in stem.split("_")])
```

Comment:
maybe simpler:
```suggestion
    name = stem.title().replace("_", "")
```
ref: https://stackoverflow.com/questions/44094703/about-python-capwords

Reply:
https://github.com/vyperlang/vyper/pull/4290/commits/d851c733d267b6f89b53628087d8313ae121b3bb